File: c/haversine.c
#include <stdio.h>
#include <stdlib.h>
#include <math.h>

#include "geotranslation.h"

int main(int argc, char** argv) {
    double lat1, lon1, lat2, lon2;
    double distance;

    if (argc != 5) {
        printf("Usage: haversine <latitude1> <longitude1> <latitute2> <longitude2>\n");
        return -1;
    }

    // grab the command-line parameters
    lat1 = atof(argv[1]);
    lon1 = atof(argv[2]);
    lat2 = atof(argv[3]);
    lon2 = atof(argv[4]);

    distance = haversine(lat1, lon1, lat2, lon2);

    printf("Distance: %f km\n", distance);
    printf("Initial bearing: %f°\n", bearingTo(lat1, lon1, lat2, lon2));
    printf("Final bearing: %f°\n", bearingFrom(lat2, lon2, lat1, lon1));

    return 0;
}

File: java/Haversine.java
public class Haversine {
    public static void main(String[] args) {
        if (args.length != 4) {
            System.out.println("Usage: example <latitude1> <longitude1> <latitute2> <longitude2>");
            return;
        }

        double lat1 = Double.parseDouble(args[0]);
        double lon1 = Double.parseDouble(args[1]);
        double lat2 = Double.parseDouble(args[2]);
        double lon2 = Double.parseDouble(args[3]);

        System.out.println("Distance: " + Geotranslation.haversine(lat1, lon1, lat2, lon2) + " km");
        System.out.println("Initial bearing: " + Geotranslation.bearingTo(lat1, lon1, lat2, lon2) + "°");
        System.out.println("Final bearing: " + Geotranslation.bearingFrom(lat2, lon2, lat1, lon1) + "°");
    }
}
